### PR TITLE
Updated Muon GUI's to usage report and deprecation of old

### LINKED ISSUE #26774: Rename muon GUIs
The original muon analysis should be renamed to muon analysis old and muon analysis 2 should be renamed as muon analysis

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**
This PR adds usage reporting to Muon Analysis, Elemental Analysis and Frequency Domain Analysis

It also renames the original Muon GUI and Muon Analysis 2.

**To test:**

Open MantidPlot
Open Muon Analysis (old) - it should be the original GUI
Open Muon Analysis - it should be the new GUI
Fixes #26774. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/usage_report.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import (absolute_import, division, unicode_literals)
+
+from mantid.kernel import UsageService
+
+def register_GUI(name):
```
Comment (NickDraper) [inline scripts/Muon/GUI/Common/usage_report.py]: Minor point but the function name here does not pass the ronseal test
Perhaps ReportInterfaceStartup(name)

File: scripts/Muon_Analysis.py [hunk 2]
```diff
@@ -8,9 +8,10 @@
 from __future__ import (absolute_import, division, print_function)
 from Muon.GUI.MuonAnalysis.muon_analysis_2 import MuonAnalysisGui
 from qtpy import QtCore
+from Muon.GUI.Common.usage_report import register_GUI
 
 
-Name = "Muon_Analysis_2"
+Name = "Muon_Analysis"
```
Comment (NickDraper) [inline scripts/Muon_Analysis.py]: Does this mean that the usage reports will be mixed with the old UI from previous versions?

We want the GUI menu items to take over the name, but internally for usage reports, we might want to keep them separate

**Review by NickDraper — CHANGES_REQUESTED** [review]
In Mantidplot your olds are not consistent
image

Reflectometry just uses "Reflectometry Old" without any brackets

File: qt/scientific_interfaces/Muon/MuonAnalysis.h [hunk 3]
```diff
@@ -60,7 +60,7 @@ class MuonAnalysis : public MantidQt::API::UserSubWindow {
 
 public:
   /// Name of the interface
-  static std::string name() { return "Muon Analysis (old)"; }
+  static std::string name() { return "Muon Analysis old)"; }
```
Comment (NickDraper) [inline qt/scientific_interfaces/Muon/MuonAnalysis.h]: Did you mean to get rid of both brackets?

File: qt/scientific_interfaces/Muon/MuonAnalysis.cpp [hunk 4]
```diff
@@ -147,8 +147,7 @@ MuonAnalysis::MuonAnalysis(QWidget *parent)
                                               // instruments later
                    {"MUSR", "HIFI", "EMU", "ARGUS", "CHRONUS"}),
       m_deadTimeIndex(-1), m_useDeadTime(true) {
-  g_log.warning(
-      "This interface is deprecated, please use Muon Analysis instead");
+  arn;
```
Comment (NickDraper) [inline qt/scientific_interfaces/Muon/MuonAnalysis.cpp]: Pretty sure this deletion was a mistake

**Review by NickDraper — CHANGES_REQUESTED** [review]

File: qt/scientific_interfaces/Muon/MuonAnalysis.cpp [hunk 5]
```diff
@@ -146,7 +146,9 @@ MuonAnalysis::MuonAnalysis(QWidget *parent)
       m_dataLoader(Muon::DeadTimesType::None, // will be replaced by correct
                                               // instruments later
                    {"MUSR", "HIFI", "EMU", "ARGUS", "CHRONUS"}),
-      m_deadTimeIndex(-1), m_useDeadTime(true) {}
+      m_deadTimeIndex(-1), m_useDeadTime(true) {
+  arn;
```
Comment (NickDraper) [inline qt/scientific_interfaces/Muon/MuonAnalysis.cpp]: arn?

**Review by NickDraper — DISMISSED** [review]

**Comment by NickDraper** [comment]
All looks good now, ready for merging once the builds pass

**Comment by NickDraper** [comment]
Flake8 needs some love

File: scripts/Muon/GUI/Common/usage_report.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import (absolute_import, division, unicode_literals)
+
+from mantid.kernel import UsageService
+
+
+def ReportInterfaceStartup(name):
+    #interface startup
+    UsageService.registerFeatureUsage("Interface", name, False)
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/usage_report.py]:
```suggestion
    UsageService.registerFeatureUsage("Interface", name, False)

```

**Review by NickDraper — DISMISSED** [review]
Reapproving after the warnings were resolved.

File: scripts/Muon/GUI/Common/usage_report.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import (absolute_import, division, unicode_literals)
+
+from mantid.kernel import UsageService
+
+
+def ReportInterfaceStartup(name):
```
Comment (DanNixon) [inline scripts/Muon/GUI/Common/usage_report.py]:
```suggestion
def report_interface_startup(name):
```

This follows Python naming conventions.

**Review by DanNixon — CHANGES_REQUESTED** [review]

File: scripts/Elemental_Analysis.py [hunk 8]
```diff
@@ -8,6 +8,7 @@
 
 from qtpy import QtCore
 from Muon.GUI.ElementalAnalysis.elemental_analysis import ElementalAnalysisGui
+from Muon.GUI.Common.usage_report import ReportInterfaceStartup
```
Comment (AnthonyLim23) [inline scripts/Elemental_Analysis.py]:
```suggestion
from Muon.GUI.Common.usage_report import report_interface_startup
```

File: scripts/Frequency_Domain_Analysis.py [hunk 9]
```diff
@@ -8,6 +8,7 @@
 from __future__ import (absolute_import, division, print_function)
 from Muon.GUI.FrequencyDomainAnalysis.frequency_domain_analysis_2 import FrequencyAnalysisGui
 from qtpy import QtCore
+from Muon.GUI.Common.usage_report import ReportInterfaceStartup
```
Comment (AnthonyLim23) [inline scripts/Frequency_Domain_Analysis.py]:
```suggestion
from Muon.GUI.Common.usage_report import report_interface_startup
```

File: scripts/Muon_Analysis.py [hunk 10]
```diff
@@ -8,6 +8,7 @@
 from __future__ import (absolute_import, division, print_function)
 from Muon.GUI.MuonAnalysis.muon_analysis_2 import MuonAnalysisGui
 from qtpy import QtCore
+from Muon.GUI.Common.usage_report import ReportInterfaceStartup
```
Comment (AnthonyLim23) [inline scripts/Muon_Analysis.py]:
```suggestion
from Muon.GUI.Common.usage_report import report_interface_startup
```

**Review by NickDraper — APPROVED** [review]
Good to go I think, The final change addressed Dan's concerns as well

**Review by DanNixon — APPROVED** [review]
RIP well formed commit history

**Pull request merged**